libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -83,19 +85,22 @@ export function loadBlock(path) {
 
 let cachedDecorateMenu;
 export async function loadDecorateMenu() {
-  // eslint-disable-next-line no-async-promise-executor
-  cachedDecorateMenu = cachedDecorateMenu || new Promise(async (resolve) => {
-    const [{ decorateMenu, decorateLinkGroup }] = await Promise.all([
-      loadBlock('./menu/menu.js'),
-      loadStyles('utilities/menu/menu.css'),
-    ]);
-
-    resolve({
-      decorateMenu,
-      decorateLinkGroup,
-    });
+  if (cachedDecorateMenu) return cachedDecorateMenu;
+
+  let resolve;
+  cachedDecorateMenu = cachedDecorateMenu || new Promise((_resolve) => {
```

Comment:
If `cachedDecorateMenu` exists, wouldn't the function have returned in `if (cachedDecorateMenu) return cachedDecorateMenu;`? So the OR might not be necessary.
```suggestion
  cachedDecorateMenu = new Promise((_resolve) => {
```